feat: add auth(both) check and redirection

d9nielcoding opened the pull request:
## Description

- Updated .gitignore to include all .env files except .env.example.
- Refactored navigation hooks to remove userCode parameters from certain navigation functions for cleaner code.
- Adjusted components to utilize the new navigation structure, enhancing code maintainability.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that may cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Affected Pages

- [x] Edit User Page
- [x] Create/Edit Idea/List Page 

## Related Project Task (Linear)

https://linear.app/relistspace/issue/PD-255/unauthenticated-listidea-createedit-profile-edit-blocking

## Checklist

- [x] I have manually tested this locally to ensure functionality and stability
- [ ] I have commented my code, especially in hard-to-understand areas
- [ ] I have updated the documentation, or my changes do not require documentation updates
- [x] My PR has a clear title and description

jamespig commented on src/hooks/mutations/useSocialAction.ts:
這隻檔案在我那個PR被砍掉了，這樣是不是會對撞？

poklist-dev replied:
那個 conflicts 本來就會擋 PR，不用特地留言 XD

d9nielcoding committed:
feat: add auth(both) check and redirection

- Updated .gitignore to include all .env files except .env.example.
- Refactored navigation hooks to remove userCode parameters from certain navigation functions for cleaner code.
- Adjusted components to utilize the new navigation structure, enhancing code maintainability.

d9nielcoding committed:
feat: add external links and refactor navigation

- Added new external links for feedback, Discord, Instagram, Threads, and LinkedIn in externalLink.ts.
- Updated Home index.en.ts to utilize the new external links.
- Refactored useAuth.ts and other hooks to improve type safety and code clarity.
- Enhanced useClipboard.ts by removing unnecessary useEffect for clipboard support detection.
- Updated various components to improve navigation and state management.
- Removed deprecated LoginErrorDialog component.
